scilpy/utils/streamlines.py

```diff
@@ -675,4 +680,11 @@ def cut_invalid_streamlines(sft):
                                           data_per_streamline=new_data_per_streamline,
                                           data_per_point=new_data_per_point)
 
+    # Move the streamlines back to the original space/origin
+    sft.to_space(space)
+    sft.to_origin(origin)
+
```

Comment:
do we need sft here ?

Reply:
Yes, we do. Changing the original space/origin to a new space/origin inside the method and not changing it back before returning leaves the sft in the new space/origin. That same sft object might be used for further processing (even if it is to use its header properties) back at the caller script, which will lead to unexpected results.

Reply:
I agree with Jon. The only question that I have is why do you revert changing the sft in the from_sft call ?

Reply:
> The only question that I have is why do you revert changing the sft in the from_sft call ?

So that is linked to [this comment](https://github.com/scilus/scilpy/pull/518/files#r741209633). The suggestion was tried in commit 09dcc55 and tests were failing: https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-518/2/. I did not dig further.

Let me know if you think something else other is required to avoid the failing tests.